src/main/resources/templates/index.html

```diff
@@ -0,0 +1,18 @@
+<!DOCTYPE html>
+<html lang="pl" xmlns:th="http://www.thymeleaf.org">
+<head>
+    <meta charset="UTF-8">
+    <title>Strona główna</title>
+    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.2/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-T3c6CoIi6uLrA9TneNEoa7RxnatzjcDSCmG1MXxSR1GAsXEV/Dwwykc2MPK8M2HN" crossorigin="anonymous">
+</head>
+<body>
+<div class="container">
+<h1>Witaj na stronie głównej</h1>
+
+<a th:href="@{/register}">Zarejestruj się</a></br>
+<a th:href="@{/login}">Zaloguj się</a></br>
```

Comment:
Te elementy pokazuj tylko jeśli użytkownik nie jest jeszcze zalogowany. Możesz to zrobić dodając atrybut: `sec:authorize="isAuthenticated()"` - jest to z biblioteki `thymeleaf-extras-springsecurity6`, którą masz już dodaną.